manim/_config/main_utils.py

```diff
@@ -150,14 +153,34 @@ def _parse_args_cfg_subcmd(args):
     return parsed
 
 
+def _parse_args_plugins(args):
+    """Parse arguments of the form 'manim plugins <args>'."""
+    parser = argparse.ArgumentParser(
+        description="Utility command for managing plugins",
+        prog="manim plugins",
+        epilog="Made with <3 by the manim community devs",
+        usage=("%(prog)s -h -l"),
```

Comment:
Is the `-u` flag you've described not defined yet? I'm trying to install a plugin but it gives the following error:
![image](https://user-images.githubusercontent.com/65204531/101025389-84413100-359b-11eb-9e34-d927f5136c25.png)

Reply:
To clarify, I'm using an editable install of my own `chanim` "extension" that I renamed to be `manim-chanim` to test whether locally installed (i.e. without uploading to PyPI) plugins work.
![image](https://user-images.githubusercontent.com/65204531/101026235-a8e9d880-359c-11eb-9da9-e00c9f1eeb00.png)

Reply:
Did you add the plugin entry point? IIRC that is needed.

Reply:
wdym

Reply:
oh, so using poetry is a necessity...

Reply:
that can be done using `setup.py` also if you need but I would recommend using poetry unless you have a C extension or plugin.

Reply:
On the side note @jsonvillanueva please document about how to do this without using poetry. Because somebody would be a C extension and say it is a plugin and we can't make them use poetry. I know it is easy to find out but for the sake of docs.

Reply:
Sorry, as discussed earlier with @leotrs the update flag was removed in this PR for faster iteration. I think the idea is to relinquish control of gitlab `manim-plugintemplate` pypi repo and have it stored under ManimCommunity/manim-plugintemplate on github (so everyone can make updates to it/manim team members can publish new releases for that).

See:
> 
> 
> I agree that the core should manage the plugins. At this point "management" seems to include discovery, installation, updating, listing. I'm still not convinced that we should inject all installed plugins on every scene. On the other hand, at this point this is harmless because there aren't any plugins to be installed 😅
> 
> I strongly suggest either: a) adding a way of disabling plugins, or b) keeping this PR to discovery/installation/updating/listing and _not_ doing the importing.

We decided on option b for this PR, later will add a more fully fledged plugin importing (enabling/disabling) system. But I can update the documentation for making

Reply:
> 
> 
> oh, so using poetry is a necessity...

No it's not. See this portion on [creating plugins ](https://manimce--784.org.readthedocs.build/en/784/installation/plugins.html#creating-plugins). To quote myself:
`The only requirement of manim plugins is that they specify an entry point with the group, "manim.plugins".` And then I link to the entry point documentation and give an example using poetry. But you can use Flit, setuptools, enscons, etc... as your build system. I just give an example with poetry.

Reply:
@kilacoda In particular, manim uses the [console_scripts](https://packaging.python.org/specifications/entry-points/#entry-points-specification) group in order to specify a way to use `manim` via CLI. In much the same way, this PR is using the group, `manim.plugins`, to specify that plugins should use this. In your specific case with [chanim](https://github.com/raghavg123/chanim), you would need a pyproject.toml (if you _want_ to use poetry), or you could do something similar to @naveen521kk with [text2svg](https://github.com/naveen521kk/text2svg/blob/main/setup.py#L103-L107) and use a setup.py file (for setuptools build system instead of poetry build system) to specify an entry point as he has done there.

Either way, chanim would need some slight refactoring to make it compatible with this plugin system. You don't need to name it manim-chanim either when publishing it to PyPI. This is only so that people can find plugins under a singular `manim-` search on PyPI. But the entrypoint is needed for manim to find/list plugins (later new functionality/flags will be added)

Reply:
Thanks for clarifying. I tried adding the `entry_point` to my setup.py as follows:
```py
    entry_points={
        "manim.plugins":[
            "chanim = chanim"
        ]
    }
```
yet `manim plugins -l` still doesn't show it as a plugin. Am I doing something wrong here? Also, I believe the documentation for creating plugins could be improved a bit. Telling people `name = object_reference` is a bit vague and confusing imo, which might easily trip up new plugin contributors. Just a thought.